### PR TITLE
class fields: fix yield/await expressions inside class scope

### PR DESCRIPTION
**Pull request by caitp**
setIsFunction() was strictly unnecessary for setIsPrivateNameScope()

### CODE REVIEW

**Comment by xanlpz** [comment]
This breaks some existing tests fwiw:

stress/class-fields-harmony.js.default: Exception: SyntaxError: new.target is not valid inside arrow functions in global code.
stress/class-fields-harmony.js.default: at class-fields-harmony.js:345
stress/class-fields-harmony.js.no-llint: Exception: SyntaxError: new.target is not valid inside arrow functions in global code.
stress/class-fields-harmony.js.no-llint: at class-fields-harmony.js:345
stress/class-fields-harmony.js.no-llint: ERROR: Unexpected exit code: 3
stress/class-fields-harmony.js.default: ERROR: Unexpected exit code: 3

**Comment by caitp** [comment]
> This breaks some existing tests fwiw:
> 
> stress/class-fields-harmony.js.default: Exception: SyntaxError: new.target is not valid inside arrow functions in global code.
> stress/class-fields-harmony.js.default: at class-fields-harmony.js:345
> stress/class-fields-harmony.js.no-llint: Exception: SyntaxError: new.target is not valid inside arrow functions in global code.
> stress/class-fields-harmony.js.no-llint: at class-fields-harmony.js:345
> stress/class-fields-harmony.js.no-llint: ERROR: Unexpected exit code: 3
> stress/class-fields-harmony.js.default: ERROR: Unexpected exit code: 3

I'ved fixed this in the second commit

File: Source/JavaScriptCore/parser/Parser.h [hunk 1]
```diff
@@ -305,9 +307,14 @@ struct Scope {
...
     void setIsPrivateNameScope()
     {
-        setIsFunction();
         m_isPrivateNameScope = true;
     }
+    
+    void setIsClassScope()
+    {
+        setIsPrivateNameScope();
+        m_isClassScope = true;
```
Comment (caitp) [inline Source/JavaScriptCore/parser/Parser.h]: This isn't strictly necessary for now (could get rid of this boolean), but AFAIK there is the potential for non-class Private Name Environments in ECMA262, and new.target / super would not make sense in some of them. So having a separate boolean for class lexical scopes makes sense for that.

BUT for now it is totally optional.

**Comment by caitp** [comment]
Last commit is an optional version that doesn't add the distinction between "class scopes" and "private name environments", so it's a bit smaller --- up to you if you prefer that version or not

**Comment by xanlpz** [comment]
This was merged manually.

**Pull request closed without merging**